jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/ExtractTarStep.java

```diff
@@ -88,14 +88,17 @@ static boolean isGzipped(Path path) throws IOException {
   private final BuildConfiguration buildConfiguration;
   private final Path tarPath;
   private final ProgressEventDispatcher.Factory progressEventDispatcherFactory;
+  private final Queue<Path> directoriesToDelete;
 
   ExtractTarStep(
       BuildConfiguration buildConfiguration,
       Path tarPath,
-      ProgressEventDispatcher.Factory progressEventDispatcherFactory) {
+      ProgressEventDispatcher.Factory progressEventDispatcherFactory,
+      Queue<Path> directoriesToDelete) {
```

Comment:
Why a `Queue` specifically instead of a `Collection` or `Set`?  (Queue's API can throw an exception on `add()`, which is unhandled below.)

Would it make sense to maintain this cleanup logic in the `BuildConfiguration`?  Post-Build Clean Up Hooks?

Reply:
No reason really, I guess we can use a `Set`.

Not sure if we should keep track of the directories to cleanup in `BuildConfiguration`, since I think `BuildConfiguration` is supposed to be immutable?

I suppose another option would be to create the temporary directories beforehand in `StepsRunner` and pass the paths into the build steps that need them so they're easier to cleanup later.

Reply:
This is simpler, but just throwing an idea that I think looks better from an API standpoint:

In a similar vain, we could turn the set to a class like `TempDirectoryProvider` that has `newDirectory()` and keeps track of created directories to delete them later? We already have `TemporaryDirectory`, but looks like it doesn't fit into this in its current form. (And perhaps `StepsRunner` will have to call `close()` in the `finally` block.)

Reply:
That does seem cleaner from an API standpoint, although I don't like adding new classes for such small use cases. I'll give it a shot regardless.